src/server/index.d.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+export interface kbn_server {
+    config: () => configObject;
+  }
+  
+export interface configObject {
+    get: (path: string) => ConfigEntry;
```

Comment:
I think ConfigEntry should be `any`.  I know we generally avoid `any` as much as possible, but here, you really can't type what will be returned unless you know what `path` is.   I think there is a way to map what the string is to expected object, but for the sake of this PR, I say just make `any` here for now, and move the below typings to the reporting folder, so you can still do `get captureConfig(): ConfigEntry {`